.github/workflows/docker.yml

```diff
@@ -0,0 +1,74 @@
+name: Create and publish a Docker images
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    name: PHP ${{ matrix.php-version }}, Fixer ${{ github.ref_name }}
+    if: github.repository == 'PHP-CS-Fixer/PHP-CS-Fixer'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - php-version: '7.4'
+            alpine-version: '3.16'
+          - php-version: '8.0'
+            alpine-version: '3.16'
+          - php-version: '8.1'
+            alpine-version: '3.18'
+          - php-version: '8.2'
+            alpine-version: '3.18'
+          - php-version: '8.3'
+            alpine-version: '3.18'
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Make image lowercase
+        run: |
+          echo IMAGE_NAME_LOWER=$(echo ${{ env.IMAGE_NAME }} | tr '[:upper:]' '[:lower:]') >> ${GITHUB_ENV}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          flavor: |
+            latest=false
+            suffix=-${{ matrix.php-version }}
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME_LOWER }}
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
```

Comment:
if i create RC tag or other, eg `v4.0.0-beta.2`, will the flow work properly, or that tag syntax will fail on one of further steps ? or maybe even override and create `4.0.0` ?

I see the github action will be trigger, but I an not familiar with docker/metadata-action. will it simply use `v4.0.0-beta.2` to create `4-php8.3` docker tag?
if yes, we need to prevent this.

Reply:
Workflow starts on tags in `v*.*.*` format, so I'm not sure whether tags with suffixes will trigger it - I would say yes 🤔. @hgiesenow?

Reply:
yes, `v*.*.*` condition will execute the GH action on `v4.0.0-beta.2` tag.

What i am no sure is how docker/metadata-action will behave for such tag.
It is a blocker if if will override `4` tag

Reply:
For now let's just [narrow the pattern for triggering workflow](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7555#discussion_r1442305419).